src/main/java/com/endava/internship/warmup/domain/service/ArrayProcessorWithForLoops.java

```diff
@@ -1,153 +1,327 @@
+        int[] arr = new int[input.length];
+        for (int i = 0; i < input.length; i++) {
+            if (input[i] % 2 == 0) {
+                arr[i] = input[i] * 2;
+            } else if (input[i] % 2 != 0) {
+                arr[i] = -input[i];
+            }
+        }
+
+        return arr;
     }
 
     /**
      * Find the second max value in the array
+     *
      * @param input non-null immutable array of ints
      */
     @Override
     public int findSecondMax(final int[] input) {
-        throw new UnsupportedOperationException("Method not implemented yet");
+        int[] fsmArr = new int[input.length];
+        for (int i = 0; i < fsmArr.length; i++) {
+            fsmArr[i] = input[i];
+        }
+        int[] fsmArr_final = Arrays.stream(fsmArr).distinct().toArray();
+        /*Arrays.sort(fsmArr);*/
+        /*int tv;
+        for(int i = 0;i< fsmArr.length;i++){
+            for(int j= i+1; j<fsmArr.length;i++){
+                if(fsmArr[i] > fsmArr[j]){
+                    tv = fsmArr[i];
+                    fsmArr[i] = fsmArr[j];
+                    fsmArr[j] = tv;
+                }
+            }
+        }*/
+        return fsmArr_final[fsmArr.length - 2];
     }
 
     /**
      * Return in reverse first negative numbers, then positive numbers from array
+     *
      * @param input non-null immutable array of ints.
      * @return example: input {3, -5, 4, -7, 2 , 9}
-     *                  result: {-7, -5, 9, 2, 4, 3}
+     * result: {-7, -5, 9, 2, 4, 3}
      */
     @Override
     public int[] rearrange(final int[] input) {
-        throw new UnsupportedOperationException("Method not implemented yet");
+        int[] arrRearranged = new int[6];
+        int minus = 0;
+        int minus_array[] = new int[2];
+        int plus_array[] = new int[4];
+        int plus = 0;
+        int j = 0;
+        int r = 0;
+        int inc = 0;
+
+        for (int i = 0; i < input.length; i++) {
+            if (input[i] < 0) {
+                minus = input[i];
+                minus_array[j] = input[i];
+                j++;
+            } else if (input[i] > 0) {
+                plus = input[i];
+                plus_array[r] = input[i];
+                r++;
+            }
+        }
+        for (int i = minus_array.length - 1; i >= 0; i--) {
+            arrRearranged[inc] = minus_array[i];
+            inc++;
+        }
+        for (int i = plus_array.length - 1; i >= 0; i--) {
+            arrRearranged[inc] = plus_array[i];
+            inc++;
+        }
+        return arrRearranged;
     }
 
     /**
      * Remove (filter) all values which are smaller than (input max element - 10)
+     *
      * @param input non-null immutable array of ints
      * @return The result array should not contain empty cells!
      */
     @Override
     public int[] filter(final int[] input) {
-        throw new UnsupportedOperationException("Method not implemented yet");
+        int[] arrFiltred = new int[4];
```

Comment:
```suggestion
        int[] arrFiltred = new int[input.length];
```
otherwise will not work if there should be more than 4 items

Reply:
Done